src/NuGet.Clients/PackageManagement.VisualStudio/EnvDTEProjectAdapter.cs

```diff
@@ -156,21 +158,27 @@ public bool IsLegacyCSProjPackageReferenceProject
 
                 if (!_isLegacyCSProjPackageReferenceProject.HasValue)
                 {
-                    // A legacy CSProj can't be CPS, must cast to VSProject4 and *must* have at least one package
-                    // reference already in the CSProj. In the future this logic may change. For now a user must
-                    // hand code their first package reference. Laid out in longhand for readability.
-                    if (AsIVsHierarchy?.IsCapabilityMatch("CPS") ?? true)
-                    {
-                        _isLegacyCSProjPackageReferenceProject = false;
-                    }
-                    else if (AsVSProject4 == null ||
-                        (AsVSProject4.PackageReferences?.InstalledPackages?.Length ?? 0) == 0)
+                    // A legacy CSProj must cast to VSProject4 to manipulate package references
+                    if (AsVSProject4 == null)
                     {
                         _isLegacyCSProjPackageReferenceProject = false;
                     }
                     else
                     {
-                        _isLegacyCSProjPackageReferenceProject = true;
+                        // Check for RestoreProjectStyle property
+                        var restoreProjectStyle = GetMSBuildProperty(AsIVsBuildPropertyStorage, RestoreProjectStyle, null);
```

Comment:
it was already there, this PR explicitly added a new method with 3 arguments and the reason is to pass configName as NULL instead of string.empty which is the default behavior as of now. We didn't want to change the existing behavior which is why this new overloaded method is added.